ddtrace/contrib/internal/openai/_endpoint_hooks.py

```diff
@@ -528,6 +532,35 @@ def _record_response(self, pin, integration, span, args, kwargs, resp, error):
         integration.llmobs_set_tags(span, args=[], kwargs=kwargs, response=resp, operation="response")
         return resp
 
+    def _trace_mcp_tool_usage(self, pin, integration, resp):
+        """Detect and trace server-side MCP tool usage in the response."""
+        if not resp:
+            return
+
+        messages = _get_attr(resp, "output", [])
+
+        if messages and isinstance(messages, list):
+            for item in messages:
+                message_type = _get_attr(item, "type", "")
+                if message_type == "mcp_call":
+                    call_id = str(_get_attr(item, "id", ""))
+                    name = str(_get_attr(item, "name", ""))
+                    raw_arguments = _get_attr(item, "arguments", OAI_HANDOFF_TOOL_ARG)
+                    arguments = safe_load_json(str(raw_arguments))
+                    output = str(_get_attr(item, "output", ""))
+                    self._create_mcp_tool_span(call_id, name, arguments, output, integration, pin)
+
+    def _create_mcp_tool_span(self, tool_id, tool_name, tool_arguments, tool_output, integration, pin):
+        """Creates and submits a tool span to LLMObs to represent a server-side MCP tool call."""
+        with integration.trace(pin, "server_tool_call", submit_to_llmobs=True, kind="tool") as span:
```

Comment:
The way this is written will mean that mcp tool spans generated from a response will have a crazy short duration. Is there anyway we can get timing information from the serverside tool use operation and set that on the tool span?

Alternatively, if we can grab the timing information and we only care about llmobs generating the span, we can avoid creating a new apm span via `integration.trace(...)` entirely and just create a new llmobs span the same way we do in the bedrock_agents integration. Up to you, but I think getting the timing information on the span is pretty important if possible